com.salesforce.b2eclipse.jdt.ls/src/main/java/com/salesforce/b2eclipse/util/IntellijAspectPackageInfoLoader.java

```diff
@@ -0,0 +1,223 @@
+package com.salesforce.b2eclipse.util;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.stream.JsonReader;
+import com.salesforce.b2eclipse.BazelJdtPlugin;
+import com.salesforce.b2eclipse.model.AspectOutputJars;
+import com.salesforce.b2eclipse.model.AspectPackageInfo;
+
+import org.apache.commons.lang3.StringUtils;
+
+public final class IntellijAspectPackageInfoLoader {
+    public static final String PROPERTY_BUILD_FILE_ARTIFACT_LOCATION = "build_file_artifact_location"; //$NON-NLS-1$
+    public static final String PROPERTY_DEPENDENCIES = "deps"; //$NON-NLS-1$
+    public static final String PROPERTY_LABEL = "label"; //$NON-NLS-1$
+    public static final String PROPERTY_TARGET = "target"; //$NON-NLS-1$
+    public static final String PROPERTY_RELATIVE_PATH = "relative_path"; //$NON-NLS-1$
+    public static final String PROPERTY_JAVA_IDE_INFO = "java_ide_info"; //$NON-NLS-1$
+    public static final String PROPERTY_JARS = "jars"; //$NON-NLS-1$
+    public static final String PROPERTY_GENERATED_JARS = "generated_jars"; //$NON-NLS-1$
+    public static final String PROPERTY_JAR = "jar"; //$NON-NLS-1$
+    public static final String PROPERTY_KEY = "key"; //$NON-NLS-1$
+    public static final String PROPERTY_SOURCE_JAR = "source_jar"; //$NON-NLS-1$
+    public static final String PROPERTY_SOURCES = "sources"; //$NON-NLS-1$
+    public static final String PROPERTY_INTERFACE_JAR = "interface_jar"; //$NON-NLS-1$
+    public static final String PROPERTY_KIND_STRING = "kind_string"; //$NON-NLS-1$
+    public static final String PROPERTY_MAIN_CLASS = "main_class"; //$NON-NLS-1$
+    public static final String PROPERTY_ROOT_EXECUTION_PATH_FRAGMENT = "root_execution_path_fragment"; //$NON-NLS-1$
+
+    private IntellijAspectPackageInfoLoader() {
+    }
+
+    public static ImmutableMap<String, AspectPackageInfo> loadAspectFiles(List<String> aspectFiles) {
+        ImmutableMap.Builder<String, AspectPackageInfo> infos = ImmutableMap.builder();
+        for (String aspectFile : aspectFiles) {
+            AspectPackageInfo buildInfo = loadAspectFile(new File(aspectFile));
+            if (buildInfo != null && buildInfo.getLabel() != null) {
+                infos.put(buildInfo.getLabel(), buildInfo);
+            }
+        }
+        return infos.build();
+    }
+
+    public static AspectPackageInfo loadAspectFile(final File aspectFile) {
+        AspectPackageInfo buildInfo = null;
+        try {
+            JsonObject jsonObject = readRootJsonObject(aspectFile);
+            buildInfo =
+                    Optional.<JsonObject>ofNullable(jsonObject).filter((JsonObject rootObject) -> isValid(rootObject))
```

Comment:
We can use `IntellijAspectPackageInfoLoader::isValid` instead of `(JsonObject rootObject) -> isValid(rootObject)`